### ISSUE TITLE
find strings

### ISSUE THREAD

**Issue by eko234** [gustavo-hms/peneira#6]
Hi, I like your plugin a lot, indeed I droped fzf in favour of peneira, it feels more comfy to me, I wanted to know if you can think of a way to do fuzzy extend/search in a "/" or "?" fashion, I think it can be tricky though, can you run peneira in a "headless" mode maybe?

**Comment by gustavo-hms** [comment]
Hi, @eko234 ! I'm glad you like the plugin! Nice to here that! =)

> a way to do fuzzy extend/search in a "/" or "?" fashion

Do you mean like the built-in fuzzy finder when we are searching something in Kakoune?

![busca](https://user-images.githubusercontent.com/1659941/138366811-efd16ebf-f709-49c6-ad5f-a42b3d962628.png)

¡Un abrazo!

**Comment by eko234** [comment]
Hola! :) (tambien hablo español), yes, just like that but with phrases instead of just words.

**Comment by eko234** [comment]
> 
> 
> Hola! :) (tambien hablo español), yes, just like that but with phrases instead of just words.

Ho sorry, I see you speak portuguese, un abrazo for you too

**Comment by gustavo-hms** [comment]
Hola, @eko234! Perdón por tardar tanto para contestarte...

Soy brasileño pero si hablo español =)

No estoy seguro de que entendí lo que quieres... ¿es algo diferente de lo hace el comando `peneira-lines`? ¿Podrías explicarme tu idea con más detalles?

**Comment by eko234** [comment]
Hola, sorry, olvide contestar el otro dia, hace poco descubri que kakoune tiene autcompletado "fuzzy" con los comandos / y ? por ejemplo, algo similar es lo que preguntaba para peneira, entiendo que eso implicaria definir alguna especie de separador aunque kakoune ya resuelve ese caso por defecto, sinembargo esto me lleva a preguntarme cuan complejo seria que peneira permitiera mostrar candidatos que ocuparan mas de una linea?

**Comment by gustavo-hms** [comment]
> kakoune tiene autcompletado "fuzzy" con los comandos / y ? por ejemplo

Sí, por eso te mostré el imagen con las sugestiones en línea de Kakoune, pues imaginaba fuera algo semejante a lo que querías.

Sin embargo, todavía no entendí qué quieres. Si no es eso del autocompletado de `/` y `?`, exactamente qué es? Podrías darme un ejemplo detallado?

**Comment by eko234** [comment]
Jaja, lo siento por no ser claro  😅, me refería de hecho si, a tener algo como las sugerencias que maneja kakoune al buscar algo, pero no en un espacio tan reducido como el que ocupan. Sin embargo entiendo que visualizar el buffer es importante en ese tipo de operación para poder orientarte. 

He entendido mejor Peneira en este tiempo y creo que hacer algo así sería relativamente sencillo utilizando los conceptos que ya has planteado en el plugin, pero pienso que es algo que el usuario puede definir perfectamente.

Con mi último comentario, me preguntaba que piensas de la idea y cuán complejo sería poder tener algo más de contexto en Peneira, como por ejemplo que al estar sobre una línea, salga una info box que es llena con un comando bash (o lua), que toma la línea cómo argumento, lo digo porque estoy usando Peneira para definir kakrc's locales en cada proyecto y tener una paleta de comandos, ha resultado muy útil y práctico, pero a veces quisiera poder leer una definición breve de lo que voy a usar.

Gracias de nuevo, espero tu respuesta.

**Comment by eko234** [comment]
Lo anterior supongo se podria definir por el usuario tambien, declarar algunos hooks y opciones para el buffer segun sea el uso, pero imaginaba la posibilidad de un tercer argumento para peneria, que fuese un bash script para determinar algun evento al estar sobre cada linea.
